src/core/Akka.Remote/DeadlineFailureDetector.cs

```diff
@@ -28,7 +28,7 @@ public DeadlineFailureDetector(TimeSpan acceptableHeartbeatPause, Clock clock =
         {
             _acceptableHeartbeatPause = acceptableHeartbeatPause;
             _acceptableHeartbeatMillis = Convert.ToInt64(acceptableHeartbeatPause.TotalMilliseconds);
-            Guard.Assert(_acceptableHeartbeatPause > TimeSpan.Zero, "acceptable-heartbeat-pause must be greater than zero");
+            if (_acceptableHeartbeatPause <= TimeSpan.Zero) throw new ArgumentException("acceptable-heartbeat-pause must be greater than zero");
```

Comment:
Ah yes, that's correct - should be an `Akka.Configuration.ConfigurationException`. Could you please change that @KadekM ?